### PR TITLE
AR Detection UI

### PR DESCRIPTION
**Pull request by AmirDehestani**
Created a new UI component for displaying the detected AR tags. For each detected AR tag, this component displays the  tag ID, the generated ArUco tag, and its pose.

Suggestion for future implementation: Add a feature that allows the user to choose the ArUco dictionary.

### CODE REVIEW

**Comment by AmirDehestani** [comment]
All bugs were fixed, and all suggested features were added. Ready to merge after review.

File: scripts/python/aruco_generate_marker.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+def main():
+    try:
+        parser = argparse.ArgumentParser(formatter_class=CustomFormatter,
+                                        description="Generate a .png image of a specified maker.")
+        parser.add_argument('--id', default=1, type=int,
+                            help='Marker id to generate')
+        parser.add_argument('--size', default=200, type=int,
+                            help='Side length in pixels')
+        parser.add_argument('--border', default=20, type=int,
```
Comment (vexandmore) [inline scripts/python/aruco_generate_marker.py]: Are you using the border argument?

File: src/app/ar-ui/ar-display/ar-display.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { ArDisplayComponent } from './ar-display.component';
+
+describe('ArDisplayComponent', () => {
+  let component: ArDisplayComponent;
+  let fixture: ComponentFixture<ArDisplayComponent>;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
```
Comment (vexandmore) [inline src/app/ar-ui/ar-display/ar-display.component.spec.ts:10]: Is this a unit test file? I'm not sure why there's references to tests in this file

File: src/app/ar-ui/ar-tag/ar-tag.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { ArTagComponent } from './ar-tag.component';
+
+describe('ArTagComponent', () => {
+  let component: ArTagComponent;
+  let fixture: ComponentFixture<ArTagComponent>;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
```
Comment (vexandmore) [inline src/app/ar-ui/ar-tag/ar-tag.component.spec.ts:10]: again, not sure what the point of this test is

File: src/app/services/ros.service.ts [hunk 4]
```diff
@@ -84,12 +94,21 @@ export class RosService {
...
   // Method to subscribe to gps topic
   subscribeToGPS(callback: (data: any) => void) {
     this.gpsSubscriber.subscribe((message: any) => {
       const coordinate = {lat : message.latitude, lng: message.longitude}
       callback(coordinate)
     });
   }
+
+  // Method to subscribe to ArUco markers
+  subscribeToArucoMarkers(callback: (data: ArucoMarker) => void) {
```
Comment (vexandmore) [inline src/app/services/ros.service.ts:107]: out of interest, why is the aruco marker subscriber not subscribed to the topic in its constructor?

**Review by vexandmore — CHANGES_REQUESTED** [review]

File: src/app/ar-ui/ar-display/ar-display.component.spec.ts (hunk 2 above)
Comment (AmirDehestani) [reply src/app/ar-ui/ar-display/ar-display.component.spec.ts:10]: Auto-generated by Angular CLI. Not needed in our case.

File: src/app/services/ros.service.ts (hunk 4 above)
Comment (AmirDehestani) [reply src/app/services/ros.service.ts:107]: Like this, any subscribeTo[Topic] method can be called by other components only when they are needed.

File: scripts/python/aruco_generate_marker.py (hunk 1 above)
Comment (AmirDehestani) [reply scripts/python/aruco_generate_marker.py]: Removed

**Pull request merged**